src/lib/npmCommand.ts

```diff
@@ -160,11 +154,35 @@ export class NpmModule {
   }
 
   public show(registry: string): NpmShowResults {
-    return NpmCommand.runNpmCmd(`show ${this.module}@${this.version}`, { registry, cliRoot: this.cliRoot });
+    const showCmd = NpmCommand.runNpmCmd(`show ${this.module}@${this.version} --json`, {
+      registry,
+      cliRoot: this.cliRoot,
+    });
+
+    // `npm show` doesn't return exit code 1 when it fails to get a specific package version
+    // If `stdout` is empty then no info was found in the registry.
+    if (showCmd.stdout === '') {
+      throw new SfdxError(`Failed to find ${this.module}@${this.version} in the registry`, 'NpmError');
+    }
+
+    try {
+      return JSON.parse(showCmd.stdout) as NpmShowResults;
+    } catch (error) {
+      throw new SfdxError(error, 'ShellParseError');
+    }
   }
 
   public pack(registry: string, options?: shelljs.ExecOptions): void {
-    NpmCommand.runNpmCmd(`pack ${this.module}@${this.version}`, { ...options, registry, cliRoot: this.cliRoot });
+    try {
+      NpmCommand.runNpmCmd(`pack ${this.module}@${this.version}`, {
+        ...options,
+        registry,
+        cliRoot: this.cliRoot,
+      });
+    } catch (err) {
+      const sfdxErr = SfdxError.wrap(err);
```

Comment:
checking my understanding: `sfdxErr` is used to get the err into a typed object?

Reply:
@mshanemc yes, `NpmCommand.runNpmCmd` throws an `SfdxError` but I still need to wrap it here again to be able to access its properties.